pkg/apis/agones/v1/gameserver.go

```diff
@@ -150,6 +151,20 @@ type GameServerSpec struct {
 	SdkServer SdkServer `json:"sdkServer,omitempty"`
 	// Template describes the Pod that will be created for the GameServer
 	Template corev1.PodTemplateSpec `json:"template"`
+	// AlphaSpec describes the alpha properties for the GameServer
+	Alpha AlphaSpec `json:"alpha,omitempty"`
+}
+
+// AlphaSpec is the alpha properties of the GameServer
+type AlphaSpec struct {
+	Players PlayersSpec `json:"players"`
+}
+
+// PlayersSpec tracks the initial player capacity, and what webhooks to send events to when there are
+// connection/disconnection events.
+type PlayersSpec struct {
+	InitialCapacity int64                          `json:"initialCapacity,omitempty"`
```

Comment:
I recall a discussion elsewhere about a concrete type vs a pointer in terms of compatibility. I want to double check that this is good as an int64 and shouldn't be a *int64.

Reply:
This is an interesting question. I believe you are referring to #1325.

So I thought about this - and my thought where the following on making it a pointer:
- This is only the initial value to populate the status with, it will never get updated. So at no point will you want to update the GameServer with a different value (or if you did, it shouldn't do anything).
- If you aren't using player tracking, it will default to 0, which will default the status value to 0 which won't impact anything because you aren't using player tracking (and would be what we would likely default metric values to, even if we do make it a pointer (i.e. nil == 0).

So I left it as a concrete type that would default to 0.

WDYT?

Reply:
> This is only the initial value to populate the status with, it will never get updated.

Is that going to be enforced? e.g. I wouldn't be able to `kubectl edit gs my-gameserver` and change this part of the spec from 10 -> 20? Would that be rejected by validation or just ignored? If I need a different player capacity is the only way to get it to create a new game server / fleet?

> If you aren't using player tracking, it will default to 0, which will default the status value to 0

That sgtm.

Reply:
> Is that going to be enforced? e.g. I wouldn't be able to kubectl edit gs my-gameserver and change this part of the spec from 10 -> 20? Would that be rejected by validation or just ignored?

I was probably going to ignore it, but we could throw a validation error. I don't have strong opinions. Happy to add to `GameServer.ValidateUpdate` a validation check if you feel it should be there.

> If I need a different player capacity is the only way to get it to create a new game server / fleet?

The current capacity it stored in the GameServer.Status. In the [design](https://github.com/googleforgames/agones/issues/1033#issuecomment-571433206), `SDK.Alpha.SetPlayerCapacity(int capacity)` to update the capacity from within the GameServer. But you could also do it from the K8's API to edit the status value.

Reply:
I think a validation error would make sense. Otherwise we ignore the new value but don't let the user know that we are ignoring it.

Reply:
SGTM. Implementing!

Reply:
Erk. Just realised - we actually don't do _any_ validation on GameServers on Update (we only validate on Create). 

See: https://github.com/googleforgames/agones/blob/master/pkg/gameservers/controller.go#L147

We have an issue for this #530

But never got deep into it, as there were a few edge cases for it.

Do we want to defer this work if/when we return to #530 ?

Reply:
That's reasonable. We should add a note to #530 and think about re-prioritizing it if this becomes a point of friction for users.

Reply:
Note added :+1: